src/flownet/parameters/_relative_permeability.py

```diff
@@ -260,30 +271,47 @@ def render_output(self) -> Dict:
         partial_gen_wo = functools.partial(gen_wo, fast_pyscal=True)
         partial_gen_og = functools.partial(gen_og, fast_pyscal=True)
 
-        if self._swof and self._sgof:
-            with concurrent.futures.ProcessPoolExecutor() as executor:
-                for i, relperm in zip(  # type: ignore[assignment]
-                    parameters, executor.map(partial_gen_wog, parameters)
-                ):
-                    str_swofs += relperm.SWOF(header=False)
-                    str_sgofs += relperm.SGOF(header=False)
-        elif self._swof:
-            with concurrent.futures.ProcessPoolExecutor() as executor:
-                for i, relperm in zip(  # type: ignore[assignment]
-                    parameters, executor.map(partial_gen_wo, parameters)
-                ):
-                    str_swofs += relperm.SWOF(header=False)
-        elif self._sgof:
-            with concurrent.futures.ProcessPoolExecutor() as executor:
-                for i, relperm in zip(  # type: ignore[assignment]
-                    parameters, executor.map(partial_gen_og, parameters)
-                ):
-                    str_sgofs += relperm.SGOF(header=False)
+        if isinstance(self._interpolation_values, pd.DataFrame):
+            for i in range(len(self._unique_satnums)):
+                if self._scal_for_interp is not None:
+                    relperm = self._scal_for_interp[i + 1].interpolate(
+                        parameters[i].get("interpolate"),
+                        parameters[i].get("interpolate gas"),
+                    )
+                    if self._swof:
+                        str_swofs += relperm.SWOF(header=False)
+                    if self._sgof:
+                        str_sgofs += relperm.SGOF(header=False)
+                    if not self._swof and not self._sgof:
+                        raise ValueError(
+                            "It seems like both SWOF and SGOF should not be generated."
+                            "Either one of the two should be generated. Can't continue..."
+                        )
```

Comment:
One can use `enumerate()` for this:

```suggestion
            for i, parameter in enumerate(self._unique_satnums):
                if self._scal_for_interp is not None:
                    relperm = self._scal_for_interp[i + 1].interpolate(
                        parameter.get("interpolate"),
                        parameter.get("interpolate gas"),
                    )
                    if self._swof:
                        str_swofs += relperm.SWOF(header=False)
                    if self._sgof:
                        str_sgofs += relperm.SGOF(header=False)
                    if not self._swof and not self._sgof:
                        raise ValueError(
                            "It seems like both SWOF and SGOF should not be generated."
                            "Either one of the two should be generated. Can't continue..."
                        )
```

Reply:
Enumerate over parameters then (even starting at index "1"). Looks better, thanks.